pkg/apis/machine/v1beta1/machine_webhook.go

```diff
@@ -0,0 +1,330 @@
+
+	if ok, err := h.webhookOperations(h, m); !ok {
+		return admission.Denied(err.Error())
+	}
+
+	return admission.Allowed("Machine valid")
+}
+
+// Handle handles HTTP requests for admission webhook servers.
+func (h *defaulterHandler) Handle(ctx context.Context, req admission.Request) admission.Response {
+	m := &Machine{}
+
+	if err := h.decoder.Decode(req, m); err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	klog.V(3).Infof("Mutate webhook called for Machine: %s", m.GetName())
+
+	if ok, err := h.webhookOperations(h, m); !ok {
+		return admission.Denied(err.Error())
+	}
+
+	marshaledMachine, err := json.Marshal(m)
+	if err != nil {
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+	return admission.PatchResponseFromRaw(req.Object.Raw, marshaledMachine)
+}
+
+func defaultAWS(h *defaulterHandler, m *Machine) (bool, utilerrors.Aggregate) {
+	klog.V(3).Infof("Defaulting AWS providerSpec")
+
+	var errs []error
+	providerSpec := new(aws.AWSMachineProviderConfig)
+	if err := yaml.Unmarshal(m.Spec.ProviderSpec.Value.Raw, &providerSpec); err != nil {
+		errs = append(
+			errs,
+			field.Invalid(
+				field.NewPath("providerSpec", "value"),
+				providerSpec,
+				err.Error(),
+			),
+		)
+		return false, utilerrors.NewAggregate(errs)
+	}
+
+	if m.GetLabels() == nil {
+		m.Labels = make(map[string]string)
+	}
+	m.Labels["machine.openshift.io/cluster-api-cluster"] = h.clusterID
+
+	if providerSpec.InstanceType == "" {
+		providerSpec.InstanceType = "m4.large"
+	}
+	if providerSpec.IAMInstanceProfile == nil {
+		providerSpec.IAMInstanceProfile = &aws.AWSResourceReference{ID: pointer.StringPtr(fmt.Sprintf("%s-%s-profile", h.clusterID, "worker"))}
+	}
+	if providerSpec.UserDataSecret == nil {
+		providerSpec.UserDataSecret = &corev1.LocalObjectReference{Name: "worker-user-data"}
+	}
+
+	if providerSpec.CredentialsSecret == nil {
+		providerSpec.CredentialsSecret = &corev1.LocalObjectReference{Name: "aws-cloud-credentials"}
+	}
+
+	if providerSpec.SecurityGroups == nil {
+		providerSpec.SecurityGroups = []aws.AWSResourceReference{
+			{
+				Filters: []aws.Filter{
+					{
+						Name:   "tag:Name",
+						Values: []string{fmt.Sprintf("%s-%s-sg", h.clusterID, "worker")},
+					},
+				},
+			},
+		}
+	}
+
+	if providerSpec.Subnet.ARN == nil && providerSpec.Subnet.ID == nil && providerSpec.Subnet.Filters == nil {
+		providerSpec.Subnet.Filters = []aws.Filter{
+			{
+				Name:   "tag:Name",
+				Values: []string{fmt.Sprintf("%s-private-%s", h.clusterID, providerSpec.Placement.AvailabilityZone)},
+			},
+		}
+	}
+
+	rawBytes, err := json.Marshal(providerSpec)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	if len(errs) > 0 {
+		return false, utilerrors.NewAggregate(errs)
+	}
+
+	m.Spec.ProviderSpec.Value = &runtime.RawExtension{Raw: rawBytes}
+	return true, nil
+}
+
+func validateAWS(h *validatorHandler, m *Machine) (bool, utilerrors.Aggregate) {
+	klog.V(3).Infof("Validating aws providerSpec")
```

Comment:
Sorry, just noticed this one too
```suggestion
	klog.V(3).Infof("Validating AWS providerSpec")
```